src/nvim/ops.c

```diff
@@ -72,6 +72,12 @@ static yankreg_T y_regs[NUM_REGISTERS];
 static yankreg_T   *y_previous = NULL; /* ptr to last written yankreg */
 
 static bool clipboard_didwarn_unnamed = false;
+
+static bool clipboard_needs_update = true;  // clipboard needs to be updated
```

Comment:
I think `clipboard_needs_update` is quite confusing, to me it sounds like it does what `clip_did_set_selection` does (the register was set internally and thus clipboard needs update). Perhaps negating it and `clipboard_delay_set` or `_copy` or something, to name it after the effect it has.